src/primitives.ts

```diff
@@ -930,180 +930,61 @@ export class Wire {
-        this.all_attributes.push(new Attribute(name, type_)); 
-      }
-    }
-  }
-
-  initialize_to_display_attributes() {
-    for (let i=0; i<this.attribute_names.length; i++) {
-      var name = this.attribute_names[i];
-      for (let j=0; j<this.all_attributes.length; j++) {
-        if (name == this.all_attributes[j]['name']) {
-          this.to_display_attributes.push(this.all_attributes[j]);
-          break;
-        }
-      }
-    }
+                        serialized['slicing_variable'],
+                        serialized['data_samples'],
+                        serialized['type_'],
+                        serialized['name']);
   }
 
-  initialize_lists() {
-    this.lists = [];
-    for (let i=0; i<this.to_display_attributes.length; i++) {
-      var value = [];
-      var name = this.to_display_attributes[i].name;
-      let type_ = this.to_display_attributes[i].type_;
-      if (type_ == 'float') {
-        let min = this.elements[0][name];
-        let max = this.elements[0][name];
-        for (let j=0; j<this.elements.length; j++) {
-          let elt = this.elements[j][name];
-          if (elt>max) {
-            max = elt;
-          }
-          if (elt<min) {
-            min = elt;
-          }
-        }
-        this.lists.push([min, max]);
-      } else if (type_ == 'color') {
-        for (let j=0; j<this.elements.length; j++) {
-          let elt_color = rgb_to_string(this.elements[j][name]);
-          if (!List.is_include(elt_color, value)) {
-            value.push(elt_color);
-          }
-        }
-        this.lists.push(value);
-      } else {
-        for (let j=0; j<this.elements.length; j++) {
-          let elt = this.elements[j][name].toString();
-          if (!List.is_include(elt, value)) {
-            value.push(elt);
-          }
-        }
-        this.lists.push(value);
-      }
-    }
-  }
+  computeTotalSample(dataSamples: Object[]): number {
+    let sumSamples: number = 0;
+    let normedRatio: number = 2*Math.PI;
 
-  initialize_point_list(elements) {
-    this.point_list = [];
-    for (let i=0; i<this.elements.length; i++) {
-      var elt0 = elements[i][this.to_display_attributes[0]['name']];
-      if (this.to_display_attributes[0]['type_'] == 'float') {
-        var cx = elt0;
-      } else if (this.to_display_attributes[0]['type_'] == 'color') {
-        cx = List.get_index_of_element(rgb_to_string(elt0), this.lists[0]);
-      } else {
-        cx = List.get_index_of_element(elt0.toString(), this.lists[0]);
-      }
-      let point = new Point2D(cx, 5, MyObject.copy(this.point_style));
-      point.index = i;
-      this.point_list.push(point);
-    }
+    dataSamples.forEach(sample => {
+      sumSamples += sample[this.slicingVariable]
+    });
+    normedRatio /= sumSamples;
+    return normedRatio
   }
 
-  definePieParts(elements){
-    let sumElements: number = 0;
-    let normedRatio: number = 2*Math.PI;
+  definePieParts(dataSamples: Object[]): void {
```

Comment:
private as it is used only in this class

Reply:
done

Reply:
Related to my last comment. definePieParts will be in PieChart constructor from now on